Microsoft.Toolkit.Uwp.UI.Controls/TabView/TabView.cs

```diff
@@ -0,0 +1,395 @@
+        }
+
+        private async void TabView_Loaded(object sender, RoutedEventArgs e)
+        {
+            Loaded -= TabView_Loaded;
+
+            // The Add button is part of our default TabActionHeader, which isn't part of our control template.
+            _tabAddButton = this.FindDescendantByName(TABADDBUTTON_NAME) as ButtonBase;
+
+            if (_tabAddButton != null)
+            {
+                _tabAddButton.Click -= AddTabButton_Click;
+                _tabAddButton.Click += AddTabButton_Click;
+            }
+
+            await Dispatcher.RunAsync(Windows.UI.Core.CoreDispatcherPriority.Low, () =>
+            {
+                // Need to set a tab's selection on load, otherwise ListView resets to null.
+                SetInitialSelection();
+
+                // Need to set the contentpresenter's content here for embedded XAML TabViewItems, otherwise never loads, platform issue?
+                if (_tabContentPresenter.Content == null)
+                {
+                    _tabContentPresenter.Content = (ContainerFromItem(SelectedItem) as TabViewItem)?.Content;
+                }
+
+                // Need to 'refresh' the ContentPresenter for ItemsSource Databound scenarios for it to render... :(
+                _tabContentPresenter.UpdateLayout();
+            });
+        }
+
+        /// <inheritdoc/>
+        protected override void ClearContainerForItemOverride(DependencyObject element, object item)
+        {
+            base.ClearContainerForItemOverride(element, item);
+        }
+
+        /// <inheritdoc/>
+        protected override DependencyObject GetContainerForItemOverride()
+        {
+            return new TabViewItem()
+            {
+                HeaderTemplate = ItemHeaderTemplate,
+                IsClosable = CanCloseTabs
+            };
+        }
+
+        /// <inheritdoc/>
+        protected override bool IsItemItsOwnContainerOverride(object item)
+        {
+            return item is TabViewItem;
+        }
+
+        /// <inheritdoc/>
+        protected override Size MeasureOverride(Size availableSize)
+        {
+            return base.MeasureOverride(availableSize);
+        }
+
+        /// <inheritdoc/>
+        protected override void OnApplyTemplate()
+        {
+            base.OnApplyTemplate();
+
+            _tabContentPresenter = GetTemplateChild(TABCONTENTPRESENTER_NAME) as ContentPresenter;
+            _tabViewContainer = GetTemplateChild(TABVIEWCONTAINER_NAME) as Grid;
+            _tabItemsPresenter = GetTemplateChild(TABITEMSPRESENTER_NAME) as ItemsPresenter;
+            _tabScroller = GetTemplateChild(TABSCROLLVIEWER_NAME) as ScrollViewer;
+
+            DragLeave += TabPresenter_DragLeave;
+            DragItemsCompleted += TabPresenter_DragItemsCompleted;
+            SizeChanged += TabView_SizeChanged;
+
+            if (_tabItemsPresenter != null)
+            {
+                _tabItemsPresenter.SizeChanged -= TabView_SizeChanged;
```

Comment:
In cases where template is updated, make sure to check if old _tabItemsPresenter is not null and unregister events (same for others)